src/matrix/storage/idb/stores/OperationStore.ts

```diff
@@ -14,23 +14,43 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 import {MIN_UNICODE, MAX_UNICODE} from "./common";
+import {Store} from "../Store";
 
-export function encodeScopeTypeKey(scope, type) {
+export function encodeScopeTypeKey(scope: string, type: string): string {
     return `${scope}|${type}`;
 }
 
+interface Operation {
+    id: string;
+    type: string;
+    scope: string;
+    userIds: string[];
+    scopeTypeKey: string;
+    roomKeyMessage: RoomKeyMessage;
```

Comment:
Something like this might do the trick:
```TypeScript
interface BaseOperation {
    id: string;
    scope: string;
    userIds: string[];
    scopeTypeKey: string;
}

type OperationType
    = { type: "share_room_key"; roomKeyMessage: RoomKeyMessage; }
    | { type: "another_type"; anotherField: string }

type Operation = BaseOperation & OperationType
```